tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,500 @@
+    private final Map<String, ContributingQuestion> questions = new HashMap<>();
+
+    private void initializeWorkingDirectories(){
+        testsDirectory = Path.of(getProject().file(CoordinateUtils.replace("tests/src/$group$/$artifact$/$version$", coordinates)).getAbsolutePath());
+        metadataDirectory = Path.of(getProject().file(CoordinateUtils.replace("metadata/$group$/$artifact$/$version$", coordinates)).getAbsolutePath());
+        this.gradlew = Path.of(getProject().file("gradlew").getAbsolutePath());
+    }
+
+    private void loadQuestions() throws IOException {
+        File questionsJson = getProject().file("tests/tck-build-logic/src/main/resources/contributing/questions.json");
+        List<Question> contributingQuestions = objectMapper.readValue(questionsJson, new TypeReference<>() {});
+        for (var question : contributingQuestions) {
+            this.questions.put(question.questionKey(), new ContributingQuestion(question.question(), question.help()));
+        }
+    }
+
+    @TaskAction
+    void run() throws IOException {
+        InteractiveTaskUtils.printUserInfo("Hello! This task will help you contributing to metadata repository." +
+                " Please answer the following questions. In case you don't know the answer on the question, type \"help\" for more information");
+
+        loadQuestions();
+
+        this.coordinates = getCoordinates();
+        InteractiveTaskUtils.closeSection();
+
+        Path coordinatesMetadataRoot = getProject().file(CoordinateUtils.replace("metadata/$group$/$artifact$", coordinates)).toPath();
+        boolean isExistingLibrary = Files.exists(coordinatesMetadataRoot);
+
+        Path testsLocation = getTestsLocation();
+        InteractiveTaskUtils.closeSection();
+
+        Path resourcesLocation = getResourcesLocation();
+        InteractiveTaskUtils.closeSection();
+
+        List<String> dockerImages = getDockerImages();
+        InteractiveTaskUtils.closeSection();
+
+        List<String> packages = getAllowedPackages();
+        InteractiveTaskUtils.closeSection();
+
+        List<Coordinates> additionalTestImplementationDependencies = getAdditionalDependencies();
+        InteractiveTaskUtils.closeSection();
+
+        // initialize project
+        initializeWorkingDirectories();
+        createStubs(isExistingLibrary);
+        updateAllowedPackages(packages, isExistingLibrary);
+
+        // generate necessary infrastructure
+        addTests(testsLocation);
+        addResources(resourcesLocation);
+        addDockerImages(dockerImages);
+        addUserCodeFilterFile(packages);
+        addAdditionalDependencies(additionalTestImplementationDependencies);
+        addAgentConfigBlock();
+
+        // run agent in conditional mode
+        collectMetadata();
+
+        // create a PR
+        boolean shouldCreatePR = shouldCreatePullRequest();
+        if (shouldCreatePR) {
+            String branch = "add-support-for-" + coordinates.toString().replace(':', '-');
```

Comment:
I would put `"add-support-for-"` into a constant field called `BRANCH_NAME_PREFIX`